### ISSUE TITLE
DAU graph misleads admins on license utilization

### ISSUE THREAD

**Issue by stirby** [coder/coder#15297]
Credit to @coadler for distilling this issue from customer feedback.

## Problem statement

The General deployment settings page ([dev link](https://dev.coder.com/deployment/general)) contains a graph of daily active users across the deployment. This graph is purely intended to measure the adoption of the product within an organization and summarize user activity. Here's what this graph looks like in our dogfood deployment:

![Image](https://github.com/user-attachments/assets/36c52165-792a-4ef6-8bb6-e2fda2dde812)

However, for enterprise deployments with a license cap, a "user limit" bar appears at the top. This misleads admins into thinking this is a way to gauge license usage over time and judge whether expansion/contraction is necessary:

![Image](https://github.com/user-attachments/assets/20cc7dd5-6a3c-4983-85c0-fe7b3741580a)

The (?) tooltip doesn't clarify the purpose of this graph to admins either, simply explaining how activity is calculated:

![Image](https://github.com/user-attachments/assets/8376988d-383e-4b74-9871-91436e6795b3)

This ambiguity is a problem for administrators. To summarize
1. The graph is only titled Active Users , which is also the terminology we use for license seats
2. Nothing mentions this being DAU, except for hovering over a data point. The tool tip only explains what a user needs to do to be considered active
3. The user limit bar on the top to me implies that this is connected to license usage

## Proposal

We should clarify that this graph is purely to measure concurrent user activity.

**Requirements**:

**Must have**:

- [x] Clearly display this as Daily Active Users
- [x] Remove the user limit bar at the top for licensed deployments
- [x] Explain in the tooltip that this is for measuring user activity and has no connection to license consumption

**Should have**:

- [ ] Add a new "Registered Users" chart as the _first graph_ on the top with the upper limit bar, with the existing active users graph below it with no upper limit bar

**Nice to have**:
- [ ] On the registered users graph, overlay "suspended" and "dormant" users. We may want to be careful here since SCIM creates a ton of initial dormant users who never log in so we would want to filter those out and only have dormant users who logged in.
- [ ] Change Daily Active Users to configurable by day, week, and month just like template insights. Weekly is much more useful IMO

**Comment by matifali** [comment]
Instead of calling it licensed utilization chart, I suggest we call **registered users** count graph. It's helpful to know how the total registered users go over time on both OSS and licensed deployments. 

We can put a horizontal line to indicate the limit for licensed deployments.

**Comment by bpmct** [comment]
> Additionally, we should consider creating a license utilization chart and adding it directly below this chart. That work will be tracked in another issue.

> Instead of calling it licensed utilization chart, I suggest we call registered users count graph. It's helpful to know how the total registered users go over time on both OSS and licensed deployments.
>
> We can put a horizontal line to indicate the limit for licensed deployments.



I just added these to this issue as **should have** and **nice to have**. We want to remove the confusion quickly, but if the _should haves_ are relatively easy we get to do that now too. If it is more difficult, we can extract to a separate issue.

**Comment by bpmct** [comment]
@BrunoQuaresma do you think it is possible to do the "must have" part at least before the release? The "should have" can follow after the release as subsequent PRs?

**Comment by BrunoQuaresma** [comment]
@bpmct sure thing. I just would need the copy for the "Explain in the tooltip that this is for measuring user activity and has no connection to license consumption".

**Comment by matifali** [comment]
> Add a new "Registered Users" chart as the first graph on the top with the upper limit bar, with the existing active users graph below it with no upper limit bar

This should be on `/deployment/general` page right?

**Comment by stirby** [comment]
I don't think we need much redesign, we can use the existing graph component for "registered users" IMO.

**Comment by BrunoQuaresma** [comment]
We said the same about the forgot password 😆 I would like to have @chrifro decide.

**Comment by chrifro** [comment]
👋 I'd suggest that we add a design polish as a "nice to have". It shouldn't block the content changes that are requested by the customer and are more time-sensitive. At the same time, we can definitely make the graph visually more appealing and help to grasp the information quickly.

Some ideas:
- use a chart design similar to [shadcn ](https://ui.shadcn.com/charts)
- we could use the design with two lines (one for active user/per day and one for total registered users). The registered user graph could include a split for "suspended" and "dormant" users in the tooltip
- extend the timeframe from the last two months to all-time data
- allow filtering for days, weeks, months, and years. Maybe show monthly by default

I could create the design early next week. Let me know if I should focus on that.


Btw, thanks for the the great issue description @stirby, makes it easy to dive into this 🙏

**Comment by mtojek** [comment]
FYI not a "must-do" anymore

**Comment by BrunoQuaresma** [comment]
@chrifro from the [shadcn example charts](https://ui.shadcn.com/charts), do you have a type preference? I think for this case, a stacked area chart could work very well. What do you think?

**Comment by matifali** [comment]
Maybe something like
[![Image](https://github.com/user-attachments/assets/d35a0c88-0f91-4188-8650-ae78b6ef8d65)
](https://v0.dev/chat/yXiy1dBFC79)

**Comment by BrunoQuaresma** [comment]
Here’s the current situation: we have not implemented a redesign for the charts or this page, so I'm focusing on making only the necessary changes. @chrifro, please let me know if you would like to redesign the component now or if you prefer to do it later.

![Image](https://github.com/user-attachments/assets/a45eb01d-5474-48c3-af21-79a60845579e)

**Comment by chrifro** [comment]
We can stick to your version for now. It's a great improvement ✨

For the colors, are you using existing colors? If not, could you use the blue and green that are defined in the new design system?
For dark theme (see [Figma](https://www.figma.com/design/WfqIgsTFXN2BscBSSyXWF8/Coder-kit?node-id=1-2&t=je1aTRTkGQiF0Xyc-1) for light theme)
blue-bg: ##082f49
blue-highlight: #38bdf8
green-bg: #052e16
green-highlight: #BBF7D0

Related issue #14780

**Comment by bpmct** [comment]
Yep, we need _a chart_ that shows the total users against the seat count. We do not have to put everything in the same chart, however.

**Comment by SasSwart** [comment]
Potential partial blocker that I need input on. In short, I need to propose that we opt for a point in time gauge instead of a graph over time for total users against seat count. At least for this release.

This is because:

We don't track history for when users change status. We can't accurately display historical license count, because we only know whether a user is dormant or active *today*. We don't know when that changed. The same goes for when users where deleted. We know whether or not they were deleted. We don't know when they were deleted.

I can get a gauge done by release. The changes that need to be made to the db schema to support historical stats for dormancy and deletion require more consideration.

cc @matifali @chrifro @mafredri

**Comment by matifali** [comment]
> Potential partial blocker that I need input on. In short, I need to propose that we opt for a point in time gauge instead of a graph over time for total users against seat count. At least for this release.

We already have a point-in-time value for registered users. See the screenshot or are you talking about something else?
![Image](https://github.com/user-attachments/assets/1549e370-4907-4e82-9d3d-f50b1ea961e8)

**Comment by SasSwart** [comment]
What you've pasted there is what I had in mind. I'm glad we already have it.
As for the requirement of graphing over time.

Do we care about registered users or about license utilization? The original problem statement above is this:
> This misleads admins into thinking this is a way to gauge license usage over time

If we just want registered users over time regardless of whether or not they are consuming a license, then I can go ahead and fulfil the requirement. If we care about license utilisation, then I need to make larger changes to the DB schema that will take time.

https://coder.com/docs/admin/users#:~:text=Dormant%20accounts%20do%20not%20count%20towards%20the%20total%20number%20of%20licensed%20seats%20in%20a%20Coder%20subscription%2C%20allowing%20organizations%20to%20optimize%20their%20license%20usage

**Comment by bpmct** [comment]
Gotcha. Let's just put a simple gauge that show something similar to what we have on the licenses page above or below the DAU chart.

**Comment by SasSwart** [comment]
Based on the conversations above, I've created this issue:
https://github.com/coder/coder/issues/15740#issue-2717016564

I created a new one because I would like to clarify priority for the remaining graph. It would be easier to track in isolation.

It specifically captures the outstanding feature and documents how to achieve it. Everything else in this current issue has been accomplished.

**Comment by matifali** [comment]
I am ok with closing this. But we can keep it open to track the nice to haves. 
cc: @stirby

**Comment by SasSwart** [comment]
Latest mockup for the next iteration:
![Image](https://github.com/user-attachments/assets/3ce81c07-5b05-43c3-aabb-e2b527c92c02)

Notes:
* A tooltip would explain the meaning of active, dormant and suspended. It would mention that only active users consume a license seat.
* I've tweaked the wording on the bottom graph to mention activity, not active users, because active users has a specific meaning, with which activity cannot be confused
* The issue mentioned a specific charting library. I stuck with the one we're currently using for visual consistency.
* As long as the green line is below the white line, the user is in a valid state. Red and gray are allowed to exceed the white line because they do not consume license seats.
* If any of the three exceeds the user limit annotation line (white), then we can switch iconography and border colors to warning orange. if active exceeds the limit, we can switch to error red
* TODO: I'd like to give the lines a background to more explicitly show that the lines are stacked.

**Comment by SasSwart** [comment]
Status update:
Backend work up for review:
https://github.com/coder/coder/pull/16019
Frontend work WIP, but visible here:
[https://github.com/coder/coder/pull/15893/commits/494f1651ed6b4fe2eb3caa4159c87413[…]ilters%5B%5D=.ts&file-filters%5B%5D=.tsx&show-viewed-files=true](https://github.com/coder/coder/pull/15893/commits/494f1651ed6b4fe2eb3caa4159c874133143d60c?file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&show-viewed-files=true)

**Comment by SasSwart** [comment]
Status update. Changes have been merged that allow the database and backend to support this feature. A frontend PR is ready, but designs have since shifted and some more work on the frontend is required. To my knowledge, @BrunoQuaresma has agreed to support here. I've co-assigned him to the issue.

**Comment by mtojek** [comment]
@SasSwart @BrunoQuaresma Can you describe the scope of remaining changes for transparency purposes? Unless they are documented somewhere else.

**Comment by BrunoQuaresma** [comment]
The design has been updated and should now follow the design outlined [here](https://www.figma.com/design/gtVchocIWPGYjzaHD2OIY7/Setting-page?node-id=63-510&t=4iTpvwQHTWukdHIO-0). Additionally, the chart is now displayed on the licenses page instead of the general section.

![Image](https://github.com/user-attachments/assets/e966485c-1cfb-4c32-8f85-4847a90a147e)

**Comment by bartekgatzcoder** [comment]
The General page is going to use the old API, but we rename the series on the UI to unconfuse the users. We also add additional explanation box.

The new API that Sas created is for the License page. In scope of the discussion we decided to use only the "Active Users" series from the API, so the "dormant" and "suspended" users will not be displayed. But the active users there are in fact the "users consuming the license seats" which is different from the "General page", where we show the daily user engagement.

To sum up:
- general page: use old API that was there before
- license page: use new API that Sas created, but only for the "active" series.
- we need to touch both of the screens with the changes, consequently
